src/main/java/io/github/bobocodebreskul/context/support/BeanDefinitionReaderUtils.java

```diff
@@ -60,35 +69,88 @@ public static String generateClassBeanName(Class<?> beanClass) {
+   * @return a list of bean constructor dependency types
    */
-  public static List<BeanDependency> getBeanDependencies(Class<?> beanClass) {
-    log.trace("Scanning class {} for @Autowire candidates", beanClass.getName());
-
-    Constructor<?>[] declaredConstructors = beanClass.getDeclaredConstructors();
-
-    validateBeanClassConstructorSize(beanClass.getName(), declaredConstructors);
+  public static List<BeanDependency> getConstructorBeanDependencies(
+      Constructor<?> beanConstructor) {
+    Objects.requireNonNull(beanConstructor, () -> {
+      log.error("Failed to get bean constructor dependencies for nullable constructor");
+      return "Bean constructor has not been specified";
+    });
+    log.trace("Scanning class {} for @Autowire candidates", beanConstructor.getDeclaringClass());
 
-    Stream<Class<?>> constructorDependenciesStream = Arrays.stream(declaredConstructors)
-        .flatMap(constructor -> Arrays.stream(constructor.getParameterTypes()));
+    return Arrays.stream(beanConstructor.getParameterTypes())
+        .map(dependency -> new BeanDependency(generateClassBeanName(dependency), dependency))
+        .toList();
+  }
 
-    Stream<Class<?>> fieldDependenciesStream = Arrays.stream(beanClass.getDeclaredFields())
-        .filter(field -> field.isAnnotationPresent(Autowired.class))
-        .map(Field::getType);
+  /**
+   * Finds and returns the initialization constructor for a given bean class. The initialization
+   * constructor is determined based on the presence of the {@link Autowired} annotation and the
+   * number of available constructors.
+   *
+   * @param beanClass the class of the bean to find the initialization constructor for.
+   * @param beanName  the name of the bean, used for logging and error handling.
+   * @return the initialization constructor for the specified bean class.
+   * @throws BeanDefinitionCreationException If there are issues with the bean's constructors, such
+   *                                         as having more than one constructor without the
+   *                                         {@link Autowired} annotation or multiple constructors
+   *                                         marked with {@link Autowired}.
+   */
+  public static Constructor<?> findBeanInitConstructor(Class<?> beanClass, String beanName) {
+    Constructor<?>[] declaredConstructors = beanClass.getDeclaredConstructors();
 
-    Stream<Class<?>> methodDependenciesStream = Arrays.stream(beanClass.getDeclaredMethods())
-        .filter(method -> method.isAnnotationPresent(Autowired.class))
-        .flatMap(method -> Arrays.stream(method.getParameterTypes()));
+    // One constructor with or without @Autowired.
+    if (declaredConstructors.length == 1) {
+      log.trace("One init constructor found and registered [{}] for bean candidate [{}]",
+          declaredConstructors[0], beanClass.getName());
+      return declaredConstructors[0];
+    } else if (declaredConstructors.length > 1) {
```

Comment:
may we not use else there just write if on the next line, or let's remove this if at all? Is it possible to have less than 1 constructor?  so otherwise there more so we do not need this check

Reply:
It is possible to have 0 constructors, error message describes it well. Interface for example doesn't have any constructor.
`else if` to `if` change it's just a matter of taste, but I don't mind to change it.